src/platform/src/main/java/org/geoserver/platform/GeoServerExtensions.java

```diff
@@ -333,10 +402,43 @@ public void onApplicationEvent(ApplicationEvent event) {
         }
     }
 
-    /** Checks the context, if null will issue a warning. */
-    static void checkContext(ApplicationContext context, String bean) {
-        if (context == null && isSpringContext) {
-            LOGGER.warning("Extension lookup '" + bean + "', but ApplicationContext is unset.");
+    /**
+     * Checks the context has been provided, if null will issue a warning.
+     *
+     * <p>Intended for checking context provided as a parameter to methods such as {@link
+     * #bean(String, ApplicationContext)} and {@link #extensionNames(Class, ApplicationContext)} as
+     * used by beans that are ApplicationContextAware.
+     *
+     * @param context Application context
+     * @param bean Extension
+     * @param isGeoServerExtensionsContext Indicate use of {@link GeoServerExtensions#context}
+     */
+    static void checkContext(
+            ApplicationContext context, String bean, boolean isGeoServerExtensionsContext) {
+        if (isGeoServerExtensionsContext) {
+            if (context == null) {
```

Comment:
Running a build the formatter should get it

Reply:
> That `isSpringContext` just guards the logging statement, does not make it throw a hard exception. Tests need to fail in case of regressions, to make them evident and avoid them.

See https://osgeo-org.atlassian.net/browse/GEOS-10238 for discussion, the alternative to this PR is much harder - insisting that every time a new extension is added that a GeoServerExtensionsHelper configured so that system integration tests do not produce these warnings.